### ISSUE TITLE
[NuGet.org Bug]: Empty dependencies list for .NET tools

### ISSUE THREAD

**Issue by scottaddie** [NuGet/NuGetGallery#10020]
### Impact

It's more difficult to complete my work

### Describe the bug

When viewing the **Dependencies** tab for a .NET tool package on nuget.org, no dependencies are listed. For example:

![image](https://github.com/NuGet/NuGetGallery/assets/10702007/9ecff1c0-3643-4604-9678-2117b169b9e2)

This is a lie, since I can navigate to the *.csproj file and clearly see the dependencies. But, I shouldn't have to dive into the source to determine which dependencies exist.

### Repro Steps

1. Navigate to https://www.nuget.org/packages/dotnet-sqlcache
2. Click the **Dependencies** tab and notice the message "This package has no dependencies"
3. Navigate to [the source code](https://github.com/dotnet/aspnetcore/blob/2b5d2b36a04f3a4a9bb20bbca38617d1cd6a3a1a/src/Tools/dotnet-sql-cache/src/dotnet-sql-cache.csproj#L17-L19) and notice the 3 dependencies listed

### Expected Behavior

The dependencies listed in the *.csproj file should be displayed on the **Dependencies** tab. In the case of dotnet-sqlcache, I'd expect to see these dependencies listed:
- Microsoft.Data.SqlClient
- Azure.Identity
- System.Private.Uri

**Comment by joelverhagen** [comment]
Yes, this is a legitimate issue. The reason this is the case today is that NuGet.org only reads the `<dependencies>` node in the .nuspec to populate this tab. For tool packages, this metadata does not exist. We would need to either change the tool authoring to add this metadata in a new/existing way or glean dependencies from another artifact inside the package. More details are here in a closed, related issue:
https://github.com/NuGet/NuGetGallery/issues/8323#issuecomment-728374870

**Comment by joelverhagen** [comment]
We could start by having a short note that says something along the lines of ".NET tools have dependencies embedded in the package.", and even have a link to this issue to gather upvotes to improve the experience.

**Comment by abdulrahmanhossam** [comment]
Hi @joelverhagen 👋, 

I would love to pick up this issue! It looks like a great first issue for me to contribute to NuGetGallery.

Based on your suggestion, I will update the Dependencies tab UI. If the package type is a .NET Tool, I will display a note saying: ".NET tools have dependencies embedded in the package." along with a link pointing back to this issue to gather upvotes.

Could you please assign this to me? Thanks!

**Comment by joelverhagen** [comment]
@abdulrahmanhossam, sure, and done :). When you have a PR ready, make sure to open it against the `dev` branch and include a screenshot in your PR description. I am guessing there will be some deliberation on the wording, but you can get the initial iteration ready and the team will discuss it when the PR is open.
